_guides/managing-oauth-clients.md

```diff
@@ -0,0 +1,209 @@
+
+**EXAMPLES:**
+
+The first example is without filtering by component ID.
+
+`curl https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients \    -u
+{EMAIL}:{APIKEY}`
+
+The second example includes filtering by component ID.
+
+`curl
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/?filter[component]={{COMPONENT_ID}}
+\    -u {EMAIL}:{APIKEY}`
+
+#### OAuth clients can also be retrieved by their own ID, using the following API request:
+
+`curl
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID} \
+-u {EMAIL}:{APIKEY}`
+
+Below are request payload parameters:
+
+| **Parameter**     | **Required** | **Description**            |
+|-------------------|--------------|----------------------------|
+| `TENANT_ID`       | yes          | The ID of the tenant       |
+| `OAUTH-CLIENT_ID` | yes          | The ID of the OAuth client |
+
+**EXAMPLE:**
+
+`curl
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID} \
+-u {EMAIL}:{APIKEY}`
+
+#### To update OAuth clients in a tenant, we will use the following API request:
+
+`PATCH
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID}`
+
+Below are request payload parameters:
+
+| **Parameter**                       | **Required** | **Description**                    |
+|-------------------------------------|--------------|------------------------------------|
+| `type`                              | yes          | The value should be `oauth-client` |
+| `attributes.client_id`              | yes          | Oauth-client ID                    |
+| `attributes.client_secret`          | yes          | Oauth-client secret                |
+| `relationships.component.data.id`   | yes          | Component ID                       |
+| `relationships.component.data.type` | yes          | The value should be `component`    |
+
+**EXAMPLE:**
+
+```
+curl https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID} \
+  -X PATCH \
+  -u {EMAIL}:{APIKEY} \
+  -H 'Content-Type: application/json' -d '
+  {  
+    "data":{  
+      "type":"oauth-client",
+      "attributes":{  
+        "client_id":"{CLIENT_ID}",
+        "client_secret":"{CLIENT_SECRET}"
+      },
+      "relationships":{  
+        "component":{  
+          "data":{  
+            "id":"{COMPONENT_ID}",
+            "type":"component"
+          }
+        }
+      }
+    }
+  }'
+  ```
+
+#### To delete an OAuth client, we will use the following API request:
+
+`DELETE
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID}
+\\`
+
+Below are request payload parameters:
+
+| **Parameter**   | **Description**            |
+|-----------------|----------------------------|
+| TENANT_ID       | The ID of the Tenant       |
+| OAUTH-CLIENT_ID | The ID of the OAuth client |
+
+**EXAMPLE:**
+
+`curl -i
+https://api.elastic.io/v2/tenants/{TENANT_ID}/oauth-clients/{OAUTH-CLIENT_ID} \
```

Comment:
Notice:
Generally component, that oauth-client references to, may be deleted. In such case oauth client can not be used (no component),  but it continues to live in our platform.
Tenant admin is able to remove it, if he want

Reply:
Done